lib/charms/grafana_agent/v0/cos_agent.py

```diff
@@ -700,10 +700,17 @@ def metrics_jobs(self) -> List[Dict]:
 
                 # Apply labels to the scrape jobs
                 for static_config in job.get("static_configs", []):
+                    topo_as_dict = topology.as_dict(excluded_keys=["charm_name"])
                     static_config["labels"] = {
                         # Be sure to keep labels from static_config
                         **static_config.get("labels", {}),
-                        **topology.label_matcher_dict,
+                        # TODO: We should add a new method in juju_topology.py
+                        # that like `as_dict` method, returns the keys with juju_ prefix
```

Comment:
Open an issue in cos-lib?
Or if you consider it in-scope then have a tandem PR?

Reply:
I've opened an issue for this here: https://github.com/canonical/cos-lib/issues/18

I would open 2 in-tandem PR here and there